SleepNumberController_App.groovy

```diff
@@ -21,20 +21,66 @@
  *  of those but leverages prior work they've done for the API calls and bed management.
  *    https://github.com/natecj/SmartThings/blob/master/smartapps/natecj/sleepiq-manager.src/sleepiq-manager.groovy
  *    https://github.com/ClassicTim1/SleepNumberManager/blob/master/FlexBase/SmartApp.groovy
+ *    https://github.com/technicalpickles/sleepyq/blob/master/sleepyq/__init__.py
  */
+//file:noinspection unused
+//file:noinspection SpellCheckingInspection
+//file:noinspection GrMethodMayBeStatic
+
+import groovy.json.*
 import groovy.transform.Field
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 import java.util.concurrent.ConcurrentLinkedQueue
+import groovy.transform.CompileStatic
 import java.util.concurrent.Semaphore
 import org.json.JSONObject
+import java.text.SimpleDateFormat
+
+@Field static final String appVersionFLD  = '3.1.9'
 
 @Field static ConcurrentLinkedQueue requestQueue = new ConcurrentLinkedQueue()
 @Field static Semaphore mutex = new Semaphore(1)
-@Field static Long lastLockTime = 0
-@Field static Long lastErrorLogTime = 0
-
-@Field final String DRIVER_NAME = "Sleep Number Bed"
-@Field final String NAMESPACE = "rvrolyk"
-@Field final String API_HOST = "prod-api.sleepiq.sleepnumber.com"
+@Field volatile static Long lastLockTime = 0L
+@Field static Long lastErrorLogTime = 0L
+
+@Field static final String sNL=(String)null
```

Comment:
does pulling all these out into fields save much?  I would've imagined the interpreter would inline most constants anyway?

Reply:
My experience is the code will create the constant then use it repeatedly.